Repeated identical conditional return type triggers TS2719 ("Two different types with this name exist") under tsgo

## Steps to reproduce

`tsconfig.json`:

```json
{
  "compilerOptions": {
    "strict": true,
    "target": "ES2022",
    "module": "ESNext",
    "moduleResolution": "Bundler",
    "noEmit": true
  }
}
```

`index.ts`:

```ts
export function cast<T>(value: T): {
  as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K;
} {
  return {
    as: <K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {
      return value as K;
    },
  };
}
```

Tested with `@typescript/native-preview@beta` (`Version 7.0.0-dev.20260421.2`) and `typescript@6.0.3`.

Workaround: extract the conditional into a single named alias so both annotation sites reference the same type symbol.

```ts
type PreserveNullability<T, K extends T> = null extends T
  ? K | null
  : undefined extends T
    ? K | undefined
    : K;

export function cast<T>(value: T): {
  as: <K extends T>() => PreserveNullability<T, K>;
} {
  return {
    as: <K extends T>(): PreserveNullability<T, K> => {
      return value as PreserveNullability<T, K>;
    },
  };
}
```

## Behavior with `typescript@6.0`

Compiles cleanly (exit 0, no diagnostics).

## Behavior with `tsgo`

```
index.ts(5,5): error TS2719: Type '<K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K' is not assignable to type '<K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K'. Two different types with this name exist, but they are unrelated.
  Type 'null extends T ? K | (T & undefined) | (T & null) : undefined extends T ? K | (T & undefined) | (T & null) : K | (T & undefined) | (T & null)' is not assignable to type 'null extends T ? K | (T & null) : undefined extends T ? K | (T & undefined) : K'.
    Type 'K | (undefined extends T ? K | (T & undefined) | (T & null) : K | (T & undefined) | (T & null)) | (T & undefined) | (T & null)' is not assignable to type 'null extends T ? K | (T & null) : undefined extends T ? K | (T & undefined) : K'.
      Type 'undefined extends T ? K | (T & undefined) | (T & null) : K | (T & undefined) | (T & null)' is not assignable to type 'null extends T ? K | (T & null) : undefined extends T ? K | (T & undefined) : K'.
        Type 'K | (T & undefined) | (T & null)' is not assignable to type 'null extends T ? K | (T & null) : undefined extends T ? K | (T & undefined) : K'.
          Type 'T & undefined' is not assignable to type 'null extends T ? K | (T & null) : undefined extends T ? K | (T & undefined) : K'.
            Type 'K | (T & undefined) | (T & null)' is not assignable to type 'K | (T & null)'.
              Type 'T & undefined' is not assignable to type 'K | (T & null)'.
                Type 'T & undefined' is not assignable to type 'T & null'.
                  Type 'T & undefined' is not assignable to type 'null'.
```

The two types in the diagnostic are written out identically. Note also that in tsgo's elaboration the type has been re-distributed (`K | (T & undefined) | (T & null)`) compared with what the user wrote (`K | null` or `K | undefined`), suggesting the conditional is being reduced differently in the two positions before identity is checked.

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -23816,7 +23816,7 @@ func (c *Checker) getTypeFromConditionalTypeNode(node *ast.Node) *Type {
 		links.resolvedType = c.getConditionalType(root, nil /*mapper*/, false /*forConstraint*/, nil)
 		if outerTypeParameters != nil {
 			root.instantiations = make(map[CacheHashKey]*Type)
-			root.instantiations[getTypeListKey(outerTypeParameters)] = links.resolvedType
+			root.instantiations[getConditionalTypeKey(outerTypeParameters, nil /*alias*/, false /*forConstraint*/)] = links.resolvedType
 		}
 	}
 	return links.resolvedType

--- a/testdata/baselines/reference/compiler/conditionalContextualReturnSubstitutionCache.symbols
+++ b/testdata/baselines/reference/compiler/conditionalContextualReturnSubstitutionCache.symbols
@@ -1,0 +1,41 @@
+//// [tests/cases/compiler/conditionalContextualReturnSubstitutionCache.ts] ////
+
+=== conditionalContextualReturnSubstitutionCache.ts ===
+// https://github.com/microsoft/typescript-go/issues/3488
+
+export function cast<T>(value: T): {
+>cast : Symbol(cast, Decl(conditionalContextualReturnSubstitutionCache.ts, 0, 0))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>value : Symbol(value, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 24))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+
+    as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K;
+>as : Symbol(as, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 36))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 3, 9))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 3, 9))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 3, 9))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 3, 9))
+
+} {
+    return {
+        as: <K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {
+>as : Symbol(as, Decl(conditionalContextualReturnSubstitutionCache.ts, 5, 12))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 6, 13))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 6, 13))
+>T : Symbol(T, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 21))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 6, 13))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 6, 13))
+
+            return value as K;
+>value : Symbol(value, Decl(conditionalContextualReturnSubstitutionCache.ts, 2, 24))
+>K : Symbol(K, Decl(conditionalContextualReturnSubstitutionCache.ts, 6, 13))
+
+        },
+    };
+}
+

--- a/testdata/baselines/reference/compiler/conditionalContextualReturnSubstitutionCache.types
+++ b/testdata/baselines/reference/compiler/conditionalContextualReturnSubstitutionCache.types
@@ -1,0 +1,28 @@
+//// [tests/cases/compiler/conditionalContextualReturnSubstitutionCache.ts] ////
+
+=== conditionalContextualReturnSubstitutionCache.ts ===
+// https://github.com/microsoft/typescript-go/issues/3488
+
+export function cast<T>(value: T): {
+>cast : <T>(value: T) => { as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K; }
+>value : T
+
+    as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K;
+>as : <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K
+
+} {
+    return {
+>{        as: <K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {            return value as K;        },    } : { as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K; }
+
+        as: <K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {
+>as : <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K
+><K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {            return value as K;        } : <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K
+
+            return value as K;
+>value as K : K
+>value : T
+
+        },
+    };
+}
+

--- a/testdata/tests/cases/compiler/conditionalContextualReturnSubstitutionCache.ts
+++ b/testdata/tests/cases/compiler/conditionalContextualReturnSubstitutionCache.ts
@@ -1,0 +1,14 @@
+// @strict: true
+// @noEmit: true
+
+// https://github.com/microsoft/typescript-go/issues/3488
+
+export function cast<T>(value: T): {
+    as: <K extends T>() => null extends T ? K | null : undefined extends T ? K | undefined : K;
+} {
+    return {
+        as: <K extends T>(): null extends T ? K | null : undefined extends T ? K | undefined : K => {
+            return value as K;
+        },
+    };
+}

